DataManagementSystem/Client/DataManager.py

```diff
@@ -495,6 +495,13 @@ def putAndRegister(self, lfn, fileName, diracSE, guid=None, path=None,
       errStr = "Supplied file does not exist."
       log.debug(errStr, fileName)
       return S_ERROR(errStr)
+    # check if a catalog enforces a non standard path
+    if not path:
+      pfn = self.fileCatalog.getPfnFromLfn(lfn, diracSE)
```

Comment:
`path` is the local path of the file you are trying to upload so I am not sure of what you are doing here

Reply:
Is it ? I might be wrong but I think the local file is called `fileName`.  Furthermore if `path` is not provided the original code gets it from the lfn, whic I believe is standard fro the DFC. I kept this behaviour, but in the case of RucioFilecatalogClient, the pfn is determined by Rucio, hence a call to `self.fileCatalog.getPfnFromLfn(lfn, diracSE)` which gets the custom pfn. This is non-standard since in this case the trailing part of the pfn is not equal to the lfn.

Reply:
actually you are right, `path` is the path on the storage, but it is not used !

Reply:
anyway, as pointed by @cserf, with whom I totally agree, we should not modify DIRAC for that. Rucio apparentlycan  adopt the DIRAC convention, so this should be the way to go. And if other VO embark in this mixed journey, then the documentation page should state that clearly

Reply:
Belle2 are in a comfortable situation where they have their own Rucio server for themselves. They can define the paths the way they want. In a multi-VO Rucio environment the server admins might want to define prefixes which do not follow the Dirac convention. This is a reason why I looked into modifying the DataManager just for the Rucio file catalog, still preserving a 'normal' behaviour. As I said today, it is at least worth looking into.

Reply:
The path convention in Rucio is configurable at the Rucio Storage Element (RSE) level. Each RSE can be tagged with a specific RSE attribute that define the path convention. That way, the RSEs used by VO that used Dirac+Rucio can be tagged to use the Dirac convention, and the other RSEs can choose whatever they want.